subsys/llext/CMakeLists.txt

```diff
@@ -2,4 +2,38 @@ if(CONFIG_LLEXT)
   zephyr_library()
   zephyr_library_sources(llext.c llext_export.c buf_loader.c)
   zephyr_library_sources_ifdef(CONFIG_LLEXT_SHELL shell.c)
+
+  # Compile a source .c file to an .llext file, using the appropriate
+  # commands and flags for the target architecture.
+  # Expects full paths to the source and target files.
+  # Additional flags can be provided via the LLEXT_C_FLAGS list.
+  function(llext_compile source target)
```

Comment:
Lets do this the right way, lets setup a target and add this as a cmake module under zephyr/cmake/modules/llext.cmake

It's absolutely needed and I appreciate the work here to add this.

Ultimately we may wish to provide some options using cmake variables as well for deciding compilation route (static vs got/plt linkage) as well as wanting to reuse as many other compilation flags as possible. All of that can live in the llext.cmake file then.

Reply:
@tejlmand any thoughts on this?